zirgen/Conversions/Typing/ZhlComponent.cpp

```diff
@@ -633,6 +638,13 @@ void LoweringImpl::gen(ExternOp ext, ComponentBuilder& cb) {
           }
           return builder.create<ZStruct::PackOp>(ext.getLoc(), ty, fields);
         })
+        .Case<ArrayType>([&](ArrayType ty) {
+          llvm::SmallVector<Value> elements;
```

Comment:
You already know how many elements you're going to need, so why not allocate them up front instead of pushing and resizing?